src/clarity-angular/popover/popover.spec.ts

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2017 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {Popover} from "./popover.js";
+import {Point} from "./popover";
+
+describe("Popover", function () {
+    let container: HTMLElement;
+    let anchor: HTMLElement;
+    let popover: HTMLElement;
+    let popoverInstance: Popover;
+
+    beforeEach(() => {
+        container = document.createElement("div");
+        container.id = "container";
+        container.style.position = "relative";
+        container.style.overflow = "scroll";
+        container.style.padding = "60px";
+
+        anchor = document.createElement("div");
+        anchor.id = "anchor";
+        anchor.style.position = "relative";
+        anchor.style.height = "150px";
+        anchor.style.width = "150px";
+
+        popover = document.createElement("div");
+        popover.id = "popover";
+        popover.style.height = "50px";
+        popover.style.width = "50px";
+        popover.style.margin = "5px";
+
+        document.body.appendChild(container);
+        document.getElementById("container").appendChild(anchor);
+        document.getElementById("anchor").appendChild(popover);
+
+        popoverInstance = new Popover(popover);
+    });
+
+    it("prevents scrolling of its first positioned container", function () {
+        popoverInstance.anchor(anchor, null, null);
+        expect(container.style.overflow).toEqual("hidden");
+    });
+
+    it("resumes scrolling of its first positioned container when destroyed", function () {
+        popoverInstance.anchor(anchor, null, null);
+        popoverInstance.destroy();
+        expect(container.style.overflow).toEqual("scroll");
+    });
+
+    it("positions the popover according to align points specified", function () {
+        // popovers above the anchor
+        popoverInstance.anchor(anchor, Point.TOP_LEFT, Point.BOTTOM_RIGHT);
+        expect(popover.style.transform).toEqual("translateX(0px) translateY(0px)");
+
+        popoverInstance.anchor(anchor, Point.TOP_LEFT, Point.BOTTOM_LEFT);
+        expect(popover.style.transform).toEqual("translateX(60px) translateY(0px)");
```

Comment:
These tests would be more resilient if the translate values were computed instead of hard-coded. If we know that we want `popover.style.height` to be 50px then can't we use computed values (adding in whatever value we use for additional margin/positioning from the component added in?

If the component changes the additional 10px to an additional anything-other-than-10px, all these tests will break. But if the values were computed, it wouldn't.

This isn't something I would worry about for the PR here. But maybe we can add a `// TODO:`?